### DISCUSSION TITLE
Finding Params for custom cleaning mode.

### DISCUSSION THREAD

**Discussion by JayChun** [Tasshack/dreame-vacuum#125] [Q&A] [answered]
Firstly, thank you for the great HA integration.
I wanted to set up multiple custom cleaning modes, but the official app only allows  one profile, so I was looking for another way.

I'm creating a script for a custom cleaning mode,
and I have some question about finding SIID, PIID, AIID for custom cleaning mode.

URL below, "params:" part seems to be the default settings for the device, and the values in brackets [ ] are settings for custom cleaning.
[https://github.com/Tasshack/dreame-vacuum/issues/44#issuecomment-1330091178](url)

And I found your comment and miio spec page on HA community,
[https://community.home-assistant.io/t/custom-component-dreame-vacuum/473026/53](url)
[https://home.miot-spec.com/spec/dreame.vacuum.r2228o](url)

My Device is Dreame L10S Ultra (Exactly Xiaomi Mijia 1S B116)
And this is the same as the vacuum in the first link.
After looking at all the information above, I'm still not sure what is my Params.

Could you please let me know why his SIID is 6, AIID is 2, PIID is 4?

**Comment by Tasshack** [comment, marked as answer]
You cannot set up multiple custom cleaning profiles, device does not support that hence the app. You can add multiple maps and set different settings for each map that way.

If you want to update the custom cleaning settings programmatically https://github.com/Tasshack/dreame-vacuum/issues/44#issuecomment-1330091178 is the only way to do that.

The ids you saw on the forum post has nothing to do with the customized cleaning settings.

**Reply by JayChun** [reply]
I thought that if I created multiple scripts, Vacuum would run according to the settings in the brackets, but apparently not. 

I needed two very simple profiles. for example,
1 script for only mop all rooms.
1 script for only sweep for all rooms.

Okay, I'll try to make multiple map and setting.
Thanks for your response.

**Reply by Tasshack** [reply]
Check this https://github.com/Tasshack/dreame-vacuum/discussions/116

**Reply by JayChun** [reply]
Sorry, I forgot one. 

1 script for only mop by sequence all rooms.
1 script for only sweep by sequence for all rooms.

As I know Sequence setting is only allowed at custom cleaning setting.

**Reply by Tasshack** [reply]
You dont need to use customized cleaning settings to do what you want. If you turn off the customized cleaning feature you can pass cleaning settings to start_cleaning service for each cleaning job. If you disable custom order than robot will follow the room order that you passed on the service parameters.

https://github.com/Tasshack/dreame-vacuum/blob/master/docs/services.md#dreame_vacuumvacuum_clean_segment

**Reply by JayChun** [reply]
I think you're probably in a different time zone, but I really appreciate the quick response!!

**Reply by cpktmpkt** [reply]
Would be great if you could integrate Cleaning mode in the clean_segment service. Then it would be much easier to write sequences for sweeping only or mopping Alter sweeping. A workflow my wife prefers…

**Reply by Tasshack** [reply]
Is it possible via the app? If not it may not be possible to send it on device too.

**Reply by Tasshack** [reply]
If i try to send cleaning mode via clean_segment to my device(Z10 Pro), it crashes.

**Reply by Tasshack** [reply]
I can give you send_command equalivent of clean_segment service parameters with PIID and SIID's so you can try on your device.

**Comment by JayChun** [comment]
My App[Mi Home] allows 2 kind of cleaning mode aa below screenshot.
 
It's allowed only one Custom cleaning mode.

![Screenshot_20230421_113614_Mi Home](https://user-images.githubusercontent.com/71179304/233534873-bb606d73-f45a-4dcb-8ef2-e9105d281144.jpg)

![Screenshot_20230421_113608_Mi Home](https://user-images.githubusercontent.com/71179304/233534913-9cea6d41-538e-433b-8ca4-9dddb5a02d7c.jpg)

**Reply by Tasshack** [reply]
I have checked the source of the app and looks like it is not possible to send cleaning mode with start area clean action. This is a limitation of the device and not the integration.

**Reply by JayChun** [reply]
Thanks for checking it out. I also tried it, and it didn't work.

Anyway, really thank you for great integration.
